src/components/ContentView/ContentView.tsx

```diff
@@ -18,11 +20,48 @@ export type ContentViewProps = {
  */
 export const ContentView: React.FC<ContentViewProps> = ({ contentData, pageTitle }) => {
     const [content, setContent] = useState<ContentData>({ ...contentData } as ContentData);
-
+    const [activeSection, setActiveSection] = useState({section: "", index: 0, numSections: 0});
+    const sections: string[] = [];
     useEffect(() => {
         if (!equal(content, contentData)) {
             setContent({ ...contentData } as ContentData)
         }
+        if (contentData && contentData[pageString]) {
+
+            
+            contentData[pageString].contentOrder!!.forEach(content => {
+                if (contentData[pageString].content!![content]!!.type == "HEADER") {
+                    sections.push(content);
+                }
+            })
+            setActiveSection({section: sections[0], index: 0, numSections: sections.length});
+            let prevscroll = 0;
+            let options = {
+                root: document.querySelector('#app-content-dev'),
+                rootMargin: '0px',
+                threshold: [0.0, 0.5, 1.0]
+            };
+
+            let observerCallback = (entries: IntersectionObserverEntry[], observer: IntersectionObserver) => {
```

Comment:
would probably be better to have this be its own function completely outside this useEffect hook as well as the component itself.